### ISSUE TITLE
HDF5 library version mismatched error with latest Tensorflow build for windowsx64

### ISSUE THREAD

**Issue by AndreaFar** [tensorflow/tensorflow#8627]
Due to a bug still present in Tensorflow 1.0.1 release for windows (see https://github.com/tensorflow/tensorflow/issues/8336)
I have installed the latest nightly build tensorflow cpu win 22-mar-2017 2.25.00 as suggested

Unfortunately with this version, now kernel crashes for a different reason and I get this error:

```
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.8.15, library is 1.8.18
      SUMMARY OF THE HDF5 CONFIGURATION
      =================================

General Information:
-------------------
                   HDF5 Version: 1.8.18
                  Configured on: 2017-03-06
                  Configured by: Visual Studio 14 2015 Win64
                 Configure mode: CMAKE 3.7.2
                    Host system: Windows-10.0.14393
              Uname information: Windows
                       Byte sex: little-endian
                      Libraries:
             Installation point: C:/Program Files/HDF5

Compiling Options:
------------------
               Compilation Mode:
                     C Compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
                         CFLAGS: /DWIN32 /D_WINDOWS /W3
                      H5_CFLAGS:
                      AM_CFLAGS:
                       CPPFLAGS:
                    H5_CPPFLAGS:
                    AM_CPPFLAGS:
               Shared C Library: YES
               Static C Library: YES
  Statically Linked Executables: OFF
                        LDFLAGS: /machine:x64
                     AM_LDFLAGS:
                Extra libraries: X:inclib-vc14-x64/zlib.lib;X:/inclib-vc14-x64/libsz.lib;X:/inclib-vc14-x64/libaec.lib
                       Archiver:
                         Ranlib:
              Debugged Packages:
                    API Tracing: OFF

Languages:
----------
                        Fortran: OFF
               Fortran Compiler:
          Fortran 2003 Compiler:
                  Fortran Flags:
               H5 Fortran Flags:
               AM Fortran Flags:
         Shared Fortran Library: YES
         Static Fortran Library: YES

                            C++: ON
                   C++ Compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
                      C++ Flags: /DWIN32 /D_WINDOWS /W3 /GR /EHsc
                   H5 C++ Flags:
                   AM C++ Flags:
             Shared C++ Library: YES
             Static C++ Library: YES

Features:
---------
                  Parallel HDF5: OFF
             High Level library: ON
                   Threadsafety: OFF
            Default API Mapping: v18
 With Deprecated Public Symbols: ON
         I/O filters (external):  DEFLATE DECODE ENCODE
                            MPE:
                     Direct VFD:
                        dmalloc:
Clear file buffers before write: ON
           Using memory checker: OFF
         Function Stack Tracing: OFF
      Strict File Format Checks: OFF

```
The script crashes when` tf.contrib.layers.convolution2d` is invoked

I'm using windows 10 x64 and I have installed all available update released for the others libraries

Here the list of packages of my setup
```
alabaster (0.7.10)
anaconda-clean (1.0)
anaconda-client (1.6.2)
anaconda-navigator (1.5)
anaconda-project (0.4.1)
appdirs (1.4.3)
argcomplete (1.8.2)
asn1crypto (0.22.0)
astroid (1.4.9)
astropy (1.3.1)
Babel (2.3.4)
backports.shutil-get-terminal-size (1.0.0)
beautifulsoup4 (4.5.3)
bitarray (0.8.1)
blaze (0.10.1)
bleach (2.0.0)
bokeh (0.12.4)
boto (2.46.1)
Bottleneck (1.2.0)
cffi (1.10.0)
chardet (2.3.0)
chest (0.2.3)
click (6.7)
cloudpickle (0.2.2)
clyent (1.2.2)
colorama (0.3.7)
comtypes (1.1.3)
conda (4.3.14)
conda-build (2.1.7)
conda-verify (2.0.0)
configobj (5.0.6)
contextlib2 (0.5.4)
cryptography (1.8.1)
cycler (0.10.0)
Cython (0.25.2)
cytoolz (0.8.2)
dask (0.14.1)
datashape (0.5.4)
decorator (4.0.11)
dill (0.2.6)
docutils (0.13.1)
dynd (c328ab7)
entrypoints (0.2.2)
et-xmlfile (1.0.1)
fastcache (1.0.2)
filelock (2.0.7)
Flask (0.12)
Flask-Cors (3.0.2)
gevent (1.2.1)
glue-core (0.10.1)
glue-vispy-viewers (0.7.2)
glueviz (0.10.1)
greenlet (0.4.12)
h5py (2.7.0)
HeapDict (1.0.0)
html5lib (0.999999999)
idna (2.5)
imagesize (0.7.1)
ipykernel (4.5.2)
ipython (5.3.0)
ipython-genutils (0.2.0)
ipywidgets (6.0.0)
isort (4.2.5)
itsdangerous (0.24)
jdcal (1.3)
jedi (0.10.0)
Jinja2 (2.9.5)
jsonschema (2.6.0)
jupyter (1.0.0)
jupyter-client (5.0.0)
jupyter-console (5.1.0)
jupyter-core (4.3.0)
lazy-object-proxy (1.2.2)
llvmlite (0.16.0)
locket (0.2.0)
lxml (3.7.3)
MarkupSafe (1.0)
matplotlib (2.0.0)
mccabe (0.6.1)
menuinst (1.4.4)
mistune (0.7.4)
mpmath (0.19)
multipledispatch (0.4.9)
nb-anacondacloud (1.2.0)
nb-conda (2.0.0)
nb-conda-kernels (2.0.0)
nbconvert (5.1.1)
nbformat (4.3.0)
nbpresent (3.0.2)
networkx (1.11)
nltk (3.2.2)
nose (1.3.7)
notebook (4.4.1)
numba (0.31.0)
numexpr (2.6.2)
numpy (1.12.1)
numpydoc (0.6.0)
odo (0.5.0)
olefile (0.44)
opencv-python (3.2.0.6)
openpyxl (2.4.5)
packaging (16.8)
pandas (0.19.2)
pandocfilters (1.4.1)
partd (0.3.7)
path.py (10.1)
pathlib2 (2.2.1)
patsy (0.4.1)
pep8 (1.7.0)
pickleshare (0.7.4)
Pillow (4.0.0)
pip (9.0.1)
pkginfo (1.4.1)
ply (3.10)
prompt-toolkit (1.0.13)
protobuf (3.2.0)
psutil (5.2.0)
py (1.4.33)
pyasn1 (0.2.3)
pycosat (0.6.2)
pycparser (2.17)
pycrypto (2.6.1)
pycurl (7.43.0)
pyflakes (1.5.0)
Pygments (2.2.0)
pylint (1.6.5)
PyOpenGL (3.1.0)
pyOpenSSL (16.2.0)
pyparsing (2.2.0)
pyreadline (2.1)
pytest (3.0.7)
python-dateutil (2.6.0)
pytz (2016.10)
pywin32 (220)
PyYAML (3.12)
pyzmq (16.0.2)
QtAwesome (0.4.4)
qtconsole (4.2.1)
QtPy (1.2.1)
requests (2.13.0)
rope-py3k (0.9.4.post1)
scikit-image (0.12.3)
scikit-learn (0.18.1)
scipy (0.19.0)
seaborn (0.7.1)
setuptools (34.3.2)
simplegeneric (0.8.1)
singledispatch (3.4.0.3)
six (1.10.0)
snowballstemmer (1.2.1)
sockjs-tornado (1.0.3)
Sphinx (1.5.3)
spyder (3.1.3)
SQLAlchemy (1.1.6)
statsmodels (0.8.0)
sympy (1.0)
tables (3.3.0)
tensorflow (1.0.1)
testpath (0.3)
toolz (0.8.2)
tornado (4.4.2)
traitlets (4.3.2)
unicodecsv (0.14.1)
wcwidth (0.1.7)
webencodings (0.5)
Werkzeug (0.12.1)
wheel (0.29.0)
widgetsnbextension (2.0.0)
win-unicode-console (0.5)
wrapt (1.10.10)
xlrd (1.0.0)
XlsxWriter (0.9.6)
xlwings (0.10.4)
xlwt (1.2.0)
```

**Comment by mrry** [comment]
(I assume you're using Jupyter/IPython.) What commands have you executed in the notebook before this error is raised?

**Comment by AndreaFar** [comment]
@mrry The script crashes when `tf.contrib.layers.convolution2d` is invoked

**Comment by mrry** [comment]
Can you try running the script using [`python -m trace`](https://docs.python.org/3.5/library/trace.html) and report back with what the last lines were before the HDF5 error? That should help to track down what dependency is causing the problem.

**Comment by AndreaFar** [comment]
@mrry this is the last line
`conv2d_layer1 = tf.contrib.layers.convolution2d(batch_immagine_float, num_outputs=32, kernel_size=(5, 5), stride=(2, 2), activation_fn=tf.nn.relu, weights_initializer=tf.random_normal_initializer(mean=0.0, stddev=1.0, seed=None, dtype=tf.float32), trainable=True)`

**Comment by mrry** [comment]
Did you get that line from running with `-m trace`? I would expect to see calls into various libraries (in particular, `import` statements for dependent libraries), and we'll need that information to determine the root cause.

**Comment by AndreaFar** [comment]
@mrry unfortunately enviroment crashes completely at this line and I'm unable to track the calls with more details, if I try to use `-m trace` console start an endless loop and I'm not even able to copy paste what appears

**Comment by mrry** [comment]
If you run `python -m trace myscript.py` from the Command Prompt, can you capture the last few lines of trace output? (Or does it *crash* the command prompt? In that case, something would have to be very broken....)

**Comment by AndreaFar** [comment]
@mrry As i have said is not possible, if I try to run the script with - m trace console goes crazy ant I'm forcet to CTRL+ALT+DEL. If I run the script without -m trace the whole python crashes but let me copy paste the error described above. Yes something is very broken

**Comment by mrry** [comment]
Perhaps you could redirect the output with `-m trace` to a file?

**Comment by AndreaFar** [comment]
problem solved removing conda hdf5 and installing a newer version

**Comment by mrry** [comment]
Glad to hear it!

**Comment by YulinWang-001** [comment]
Removing and reinstalling wont work cause in this case, it says 'There is no module called 'Tensorflow', but when I install tensoeflow, the hdf5 is automatically back, so it's like a dead loop.

**Comment by gothboiclique** [comment]
> Removing and reinstalling wont work cause in this case, it says 'There is no module called 'Tensorflow', but when I install tensoeflow, the hdf5 is automatically back, so it's like a dead loop.

I know this is extremely late. But after you have installed tensorflow downgrade your h5py, the previous version is working for me.

"pip install h5py==2.9.0" hope this works for you too! :)
